src/components/common/QRCode.tsx

```diff
@@ -0,0 +1,35 @@
+import QR from 'react-qr-code';
+
+import { Box, Modal } from '@mui/material';
+
+type Props = {
+  value: string;
+  open: boolean;
+  handleClose: () => void;
+};
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: '30%',
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  p: 4,
+};
+
+const QRCode = ({ value, open, handleClose }: Props): JSX.Element => (
+  <Modal open={open} onClose={handleClose}>
```

Comment:
It's best to use the Dialog component https://mui.com/material-ui/react-dialog/
instead of the bare Modal.